### PR TITLE
Fix static topic test, await the client metadata to be refreshed

### PR DESCRIPTION
**Pull request by dongeforever**
**Make sure set the target branch to `develop`**

## What is the purpose of the change

Fix the unstable test StaticTopicIT

### CODE REVIEW

**Comment by lizhanhui** [comment]
This pull request is terrible... Magic amount of sleep duration is the root cause of random test failures. Should get rid of them.

[Stackoverflow has a thread discussing similar issues.](https://stackoverflow.com/questions/36283334/how-to-avoid-thread-sleep-in-unit-tests) Maybe we should turn to [awaitility](https://github.com/awaitility/awaitility) as recommended.

**Review by hzh0425 — APPROVED** [review]

**Review by Kvicii — APPROVED** [review]

File: test/src/test/java/org/apache/rocketmq/test/statictopic/StaticTopicIT.java [hunk 1]
```diff
@@ -117,7 +116,7 @@ public void testCommandsWithBrokers() throws Exception {
         {
             Set<String> brokers = ImmutableSet.of(broker2Name);
             MQAdminTestUtils.remappingStaticTopicWithCommand(topic, brokers, null, nsAddr);
-            Thread.sleep(500);
```
Comment (lizhanhui) [inline test/src/test/java/org/apache/rocketmq/test/statictopic/StaticTopicIT.java:120]: Fix the magic number, please.

**Review by lizhanhui — CHANGES_REQUESTED** [review]

**Comment by dongeforever** [comment]
@lizhanhui  I agree with you, await is better than sleeping.

This test is stable on local mac, but it is not stable on the Travis-ci cloud. And I have not found the root cause yet. Could you help to find it, and show the await condition in detail? I am struggling with it for days.

**Comment by dongeforever** [comment]
The root cause seems to be that the client does not refresh the metadata in time.

File: test/src/test/java/org/apache/rocketmq/test/statictopic/StaticTopicIT.java (hunk 1 above)
Comment (dongeforever) [reply test/src/test/java/org/apache/rocketmq/test/statictopic/StaticTopicIT.java:120]: done

File: test/src/test/java/org/apache/rocketmq/test/statictopic/StaticTopicIT.java [hunk 2]
```diff
@@ -340,7 +340,7 @@ public void testDoubleReadCheckConsumerOffset() throws Exception {
             sendMessagesAndCheck(producer, targetBrokers, topic, queueNum, msgEachQueue, (i + 1) * TopicQueueMappingUtils.DEFAULT_BLOCK_SEQ_SIZE);
         }
 
-        TestUtils.waitForSeconds(20);
```
[the remark this answers is not in the document]
Comment (dongeforever) [reply test/src/test/java/org/apache/rocketmq/test/statictopic/StaticTopicIT.java:343]: done

**Pull request merged**